cmd/juju/controller/destroy.go

```diff
@@ -184,9 +184,21 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 	store := c.ClientStore()
-	if !c.assumeYes {
+
+	if !c.assumeNoPrompt {
+		assumeNoPrompt, skipErr := jujucmd.CheckSkipConfirmEnvVar()
+		if skipErr != nil && !errors.Is(skipErr, errors.NotFound) {
+			return errors.Trace(skipErr)
+		} else {
+			c.assumeNoPrompt = assumeNoPrompt
+		}
+	}
```

Comment:
The deprecation warning can't go in Init, but this if block can and imo should

Reply:
In fact, if you put this in destroyControllerBase it will also be inherited by kill-controller

Reply:
Agreed, fixed